src/main/java/core/basesyntax/service/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.DataBase;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit, quantity");
+        for (Map.Entry<String, Integer> entry : DataBase.storage.entrySet()) {
+            report.append("\n");
+            report.append(entry.getKey()).append(",").append(entry.getValue());
```

Comment:
Direct access to the 'storage' map in the 'DataBase' class violates encapsulation principles. Consider creating a method in the 'DataBase' class to retrieve this data.